java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/common/PostgresValidator.java

```diff
@@ -652,55 +666,144 @@ public void close() throws Exception {
         }
     }
 
-    private boolean isDataTypeCompatible(String pgDataType, Data.DataType.TypeName typeName) {
+    private boolean isDataTypeCompatible(
+            String pgDataType,
+            Data.DataType.TypeName typeName,
+            Long charMaxLength,
+            String udtName) {
         int val = typeName.getNumber();
         switch (pgDataType) {
+            case "boolean":
+                // BOOLEAN -> BOOLEAN
+                return val == Data.DataType.TypeName.BOOLEAN_VALUE;
+            case "bit":
+                // The bit type needs to be judged together with character_maximum_length
+                if (charMaxLength == null || charMaxLength == 1) {
+                    // bit(1) -> BOOLEAN
+                    return val == Data.DataType.TypeName.BOOLEAN_VALUE;
+                } else {
+                    // bit(n>1) is not supported
+                    return false;
+                }
             case "smallint":
-                return Data.DataType.TypeName.INT16_VALUE <= val
-                        && val <= Data.DataType.TypeName.INT64_VALUE;
+            case "smallserial":
+                // SMALLINT, SMALLSERIAL -> SMALLINT
+                return val == Data.DataType.TypeName.INT16_VALUE;
```

Comment:
Is it banned to map int16 into a int32 column?

Reply:
Yes.